parser/helper/helpers.go

```diff
@@ -215,3 +224,26 @@ func (h *Helper) isAnyAddressOfType(_ context.Context, addresses []address.Addre
 	}
 	return false, nil
 }
+
+func (h *Helper) GetActorNameFromCid(actorCode cid.Cid) (string, error) {
+	actorName, err := h.GetFilecoinLib().BuiltinActors.GetActorNameFromCid(actorCode)
+	if err != nil && !strings.EqualFold(actorName, unknownActorName) {
+		return actorName, err
+	}
+
+	if strings.EqualFold(actorName, unknownActorName) {
+		return h.getActorNameFromMap(actorCode), nil
+	}
+
+	return actorName, nil
+}
+
+func (h *Helper) getActorNameFromMap(actorCode cid.Cid) string {
+	actorCid := actorCode.String()
+
+	if name, ok := actorNamesByCidMap[actorCid]; ok {
```

Comment:
Why did I do this? Because something changed in version 1.29 that for certain cid's it no longer returns the type, it returns 'unknown'

Reply:
You mean on our tests? The thing here is that if we do not understand what it is happening, we may suffer from this without knowing on other places